### PR TITLE
[create-debate][T5] Replace-flow integration (#207)

### LINKED ISSUE #207: [create-debate][T5] Replace-flow integration
## Objective

Implement T5 from the Gate 4 architecture plan: wire the active-header New Debate trigger into replace mode, preserve the current debate until save succeeds, and support explicit Cancel.

## Scope

- Connect the active-header overflow trigger to replace mode
- Keep the existing active debate intact until a replace submit succeeds
- Support explicit Cancel that returns to the current debate unchanged
- Reset the replaced debate to a fresh topic with zero carried-over arguments

## Acceptance Criteria

- [ ] New Debate enters replace mode from the active header overflow trigger
- [ ] Replace submit atomically overwrites topic and clears arguments only after persistence succeeds
- [ ] Cancel exits replace mode with the current debate fully intact
- [ ] No standalone clear action is introduced
- [ ] Inline warning behavior matches the Gate 3 and Gate 4 contract with no blocking dialog

## Traceability

- Slice path: `docs/slices/create-debate/`
- Architecture reference: `docs/slices/create-debate/05-architecture.md` — Section 2.5 (Replace mode), Section 5 (Verification Strategy), Section 6 (Task Decomposition, T5)
- Acceptance criteria: AC-34, AC-35, AC-36, AC-40
- Slice tracker: #202

## Notes

Likely files: `src/components/DebateScreen.tsx`, `src/styles/debate-screen.css`, `tests/components/DebateScreen.test.tsx`
Likely tests: replace-flow component and integration coverage
Dependencies: T3 and T4
Parallelism: sequential only

### PR DESCRIPTION
**Pull request by nishantnaagnihotri**
Closes #207

## Implementation Summary
- Wired the active-header **New Debate** action to open replace mode instead of clearing the active debate immediately.
- Kept the current debate and its arguments untouched while replace mode is open and until replace persistence succeeds.
- Added explicit replace-cancel behavior that returns to the current active debate unchanged.
- Kept replace submit behavior atomic by persisting a fresh topic with zero carried-over arguments before switching back to active mode.
- Preserved inline warning behavior in replace mode, with no blocking dialog and no standalone clear action.
- Restored keyboard focus to the header action after Cancel and hardened the focus-restore signal so it only applies to cancel transitions.
- Restored dark-theme ReplaceFlow fidelity by rendering **Cancel** with the brand token (`--color-brand-primary`), matching the approved lavender dark frame.
- Added a `name` attribute on the replace topic input to address browser form-field metadata warnings.

## Files Changed
- `src/components/ActiveDebateHeader.tsx` — overflow menu and cancel-return focus restoration wiring.
- `src/components/DebateScreen.tsx` — replace-flow state integration, cancel handling, and active/replace render gating.
- `src/components/DebateTopicForm.tsx` — replace/create topic form behavior and topic-input `name` attribute.
- `src/styles/components/debate-topic-form.css` — Cancel text color token alignment for dark-theme fidelity.
- `tests/components/DebateScreen.test.tsx` — AC-34/35/40 replace-flow integration coverage.
- `tests/components/DebateTopicForm.test.tsx` — topic-input attribute coverage and token-usage assertion for Cancel color.

## Verification Evidence
- `npm run build`
- `npm run test`
- `npm run test -- tests/components/DebateTopicForm.test.tsx tests/components/DebateScreen.test.tsx`
- `npm run test:bdd` *(known pre-existing 4 failures in `features/post-tark-vitark.feature` unchanged: scenarios at lines 33, 40, 54, 73)*

## BDD Evidence
- AC-34 mapped to `tests/components/DebateScreen.test.tsx`:
  - `enters replace mode from New Debate with inline warning`
  - `replace submit writes a fresh topic and clears carried arguments`
  - `failed replace submit keeps the current debate intact until save succeeds`
- AC-35 mapped to `tests/components/DebateScreen.test.tsx`:
  - `cancel exits replace mode with the current debate unchanged`
- AC-40 mapped to `tests/components/DebateScreen.test.tsx` and `tests/components/DebateTopicForm.test.tsx`:
  - inline warning renders in replace mode and no blocking dialog appears.

## Runtime QA Verdict Package (Gate 5.5)
- Runtime QA Verdict: **Pass**
- Validated PR head: `b4d2e5e1773cd21b33a99eb2792b5c643ab2c978`
- Route list: `/`
- AC journey mapping executed: AC-34 (replace/start), AC-35 (replace/cancel), AC-40 (inline warning/no dialog)
- Coverage matrix executed: `360x800` + `768x1024`, Light and Dark
- Figma references used for fidelity checks: `947:476`, `947:521`, `897:462`, `897:475`, `977:481`, `898:463`, `898:475`, `977:483`
- Critical fidelity result: Dark ReplaceFlow Cancel rendered lavender (`#BBC3FF`) and not dark blue
- Non-blocking note: `GET /favicon.ico` 404
- Runtime QA gate recommendation: **proceed to Gate 6**

## Review Notes
- Latest Copilot review on current head (`b4d2e5e1773cd21b33a99eb2792b5c643ab2c978`) generated no new comments.

## Agent Provenance
run-id: f027694f-059d-4efe-92c8-0188b6643b1a
task-id: create-debate/gate5/dev-207-runtime-fix
role: dev
dispatched: 2026-04-28T12:25:21.807Z
model: gpt-5.3-codex

### CODE REVIEW

File: src/components/DebateScreen.tsx [hunk 1]
```diff
@@ -93,35 +94,33 @@ export function DebateScreen() {
...
+        setIsReplaceFlowOpen(true);
         setIsFabExpanded(false);
         setIsSheetOpen(false);
         ignoreNextSheetCloseRef.current = false;
     }
 
+    function handleCancelReplaceFlow(): void {
+        setDebateActionError(null);
+        setIsReplaceFlowOpen(false);
+    }
```
[the remark this answers is not in the document]
Comment (nishantnaagnihotri) [reply src/components/DebateScreen.tsx:116]: Disposition: **Accept**. I added explicit overflow-trigger focus restoration when replace cancel returns to active mode by passing a restore-focus intent from `DebateScreen` to `ActiveDebateHeader`, focusing the overflow trigger on remount, and clearing the restore flag after focus is applied. I also added a `DebateScreen` test assertion to verify that the overflow trigger regains focus after Cancel.

File: src/components/DebateScreen.tsx [hunk 2]
```diff
@@ -188,7 +193,15 @@ export function DebateScreen() {
                             {debateActionError}
                         </p>
                     ) : null}
-                    <DebateTopicForm mode="create" onStartDebate={handleStartDebate} />
+                    {isReplaceMode ? (
+                        <DebateTopicForm
+                            mode="replace"
+                            onStartDebate={handleStartDebate}
+                            onCancelReplace={handleCancelReplaceFlow}
+                        />
```
[the remark this answers is not in the document]
Comment (nishantnaagnihotri) [reply src/components/DebateScreen.tsx:203]: Disposition: **Accept**. I made the empty-state region label mode-aware: `Create debate` in create mode and `Replace debate` in replace mode, and added coverage in `DebateScreen` tests for the replace-region label.

File: src/components/ActiveDebateHeader.tsx [hunk 3]
```diff
@@ -62,6 +66,20 @@ export function ActiveDebateHeader({
...
+        }
+
+        window.requestAnimationFrame(() => {
+            const overflowTrigger = overflowTriggerRef.current;
+            if (overflowTrigger && document.contains(overflowTrigger)) {
+                overflowTrigger.focus();
+            }
+            onOverflowFocusRestored?.();
+        });
+    }, [restoreOverflowFocus, onOverflowFocusRestored]);
```
[the remark this answers is not in the document]
Comment (nishantnaagnihotri) [reply src/components/ActiveDebateHeader.tsx]: Disposition: **Accept**. I updated the restore-focus effect to keep the RAF id and cancel it in cleanup, while also guaranteeing `onOverflowFocusRestored` runs if the callback is aborted before execution. This prevents a stuck restore flag while keeping the overflow-trigger focus restoration behavior.

File: tests/components/DebateScreen.test.tsx [hunk 4]
```diff
@@ -274,7 +274,7 @@ describe('DebateScreen', () => {
         expect(screen.queryByRole('navigation', { name: 'Debate sides legend' })).not.toBeInTheDocument();
     });
 
-    it('AC-34: uses the overflow trigger to enter a new debate flow', async () => {
+    it('AC-34, AC-40: uses the overflow trigger to enter replace mode with inline warning', async () => {
```
[the remark this answers is not in the document]
Comment (nishantnaagnihotri) [reply tests/components/DebateScreen.test.tsx]: Disposition: **Accept**. I renamed the test to domain wording (`enters replace mode from New Debate with inline warning`) while preserving AC mapping.

File: src/components/ActiveDebateHeader.tsx [hunk 5]
```diff
@@ -62,6 +66,29 @@ export function ActiveDebateHeader({
...
+            }
+            didRestoreOverflowFocus = true;
+            onOverflowFocusRestored?.();
+        });
+
+        return () => {
+            window.cancelAnimationFrame(restoreOverflowFocusAnimationFrameId);
+            if (!didRestoreOverflowFocus) {
+                onOverflowFocusRestored?.();
+            }
```
[the remark this answers is not in the document]
Comment (nishantnaagnihotri) [reply src/components/ActiveDebateHeader.tsx]: Disposition: **Accept**. I replaced the boolean restore flag with a monotonic `restoreOverflowFocusRequest` signal from `DebateScreen` to `ActiveDebateHeader`. The header effect now keys off request-id changes and only schedules/cancels RAF focus work, so StrictMode cleanup no longer clears parent state or suppresses the subsequent focus-restoration pass.

File: src/components/DebateScreen.tsx [hunk 6]
```diff
@@ -93,35 +95,35 @@ export function DebateScreen() {
...
-
+    function handleStartReplaceFlow(): void {
         setDebateActionError(null);
-        setActiveDebate(emptyActiveDebate());
-        setLocalPosts([]);
-        setSelectedSide('tark');
+        setIsReplaceFlowOpen(true);
         setIsFabExpanded(false);
         setIsSheetOpen(false);
         ignoreNextSheetCloseRef.current = false;
```
[the remark this answers is not in the document]
Comment (nishantnaagnihotri) [reply src/components/DebateScreen.tsx:109]: Disposition: **Accept**. I scoped the restore signal to the cancel transition by resetting `restoreOverflowFocusRequest` back to `0` when entering replace flow and after successful replace submit. That prevents unrelated future header remounts from re-triggering focus restoration.
